Table view controller integration

sorrymother opened the pull request.

sorrymother committed:
tableView

sorrymother committed:
pohui

sorrymother committed:
cocoaPodsIntegrate

sorrymother committed:
delDsStore

sorrymother committed:
delDsStore

KMNervanin committed:
Merge pull request #3 from KMNervanin/snapKitIntegration

Snap kit integration

sorrymother committed:
add label

sorrymother committed:
first tableView integration programmatically

KMNervanin commented on RestaurantsApp/ViewController.swift:
```diff
@@ -6,14 +6,53 @@
+        "Pizza1", "Pizza2", "Pizza3", "Pizza4", "Pizza5", "Pizza6", "Pizza7", "Pizza8", "Pizza9", "Pizza10", "Pizza11", "Pizza12", "Pizza13", "Pizza14", "Pizza15", "Pizza16", "Pizza17", "Pizza18", "Pizza19", "Pizza20", "Pizza21", "Pizza22", "Pizza23", "Pizza24", "Pizza25", "Pizza26", "Pizza27", "Pizza28", "Pizza29", "Pizza30"
+    ]
+    
+    let label = UILabel()
```
удаляем(да я знаю что это я добавлял)

KMNervanin commented on RestaurantsApp/ViewController.swift:
```diff
@@ -6,14 +6,53 @@
+    }
+    
+    func createTable() {
+        self.tableView = UITableView(frame: view.bounds, style: .plain)
```
Этот инициализатор таблицы устнавливает баундс равный родительской вью, от чего верстка становится не нужна. Теперь удаляем 38 строку и делаем верстку через snapKit

KMNervanin commented on RestaurantsApp/ViewController.swift:
```diff
@@ -6,14 +6,53 @@
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let ident = "Pizza"
```
у Вас есть на 21 строке 
let identifire = "Pizza"
его можно переиспользовать и на 54-ой строке. 
И лучше сделать одной строкой 
let cell = tableView.dequeueReusableCell(withIdentifier: identifire, for: indexPath)
чем делать то что у Вас на строках 53, 54, 55

sorrymother committed:
fix merge request comments

deleted label at 17
deleted frame & style tableView at 36
created constraints(layout) with snapKit at 43-46
changed options cell at 54.

sorrymother committed:
helpCustomCell

sorrymother committed:
customCell

KMNervanin commented on RestaurantsApp/Cells/CustomTableViewCell.swift:
```diff
@@ -7,10 +7,32 @@
-    @IBOutlet weak var customImageView: UIImageView!
-    
-    @IBOutlet weak var pizzaLabel: UIStackView!
+class PizzaModel {
```
сделайте отдельный файл для модели.

KMNervanin commented on RestaurantsApp/Cells/CustomTableViewCell.swift:
```diff
@@ -7,10 +7,32 @@
-    
-    @IBOutlet weak var pizzaLabel: UIStackView!
+class PizzaModel {
+    let name:String?
```
отступы после двоеточия(на 15 строке)
let name: String
и почему они опциональные?

KMNervanin commented on RestaurantsApp/Cells/CustomTableViewCell.swift:
```diff
@@ -7,10 +7,32 @@
+    }
 }
+
+    class CustomTableViewCell: UITableViewCell {
```
отступы!
И название ячейки. Custom - ни о чем не говорит. Это ячейка для пиццы

KMNervanin commented on RestaurantsApp/Cells/CustomTableViewCell.swift:
```diff
@@ -7,10 +7,32 @@
+        
+        @IBOutlet weak var descriptionPizza: UILabel!
+        
+        override func awakeFromNib() {
```
этот метод не нужен

KMNervanin commented on RestaurantsApp/Cells/CustomTableViewCell.swift:
```diff
@@ -7,10 +7,32 @@
+            super.awakeFromNib()
+        }
+
+        func configure(for PizzaModel: PizzaModel) {
```
где конфигурация картинки?

KMNervanin commented on RestaurantsApp/ViewController.swift:
```diff
@@ -49,13 +62,17 @@ class ViewController: UIViewController, UITableViewDataSource, UITableViewDelega
-        cell.textLabel?.text = pizzasNames[indexPath.row]
+        let pizzaModel = pizza[indexPath.row]
+        cell.configure(for: pizzaModel)
+        cell.customImageView.image = UIImage(named:pizzaModel.image ?? "2")
```
т.е. Вы сделали метод конфигурации, куда отдали модель. Но картинку воткнули тут?

KMNervanin commented on RestaurantsApp/ViewController.swift:
```diff
@@ -9,13 +9,26 @@ import UIKit
-    let pizzasNames = [
-        "Pizza1", "Pizza2", "Pizza3", "Pizza4", "Pizza5", "Pizza6", "Pizza7", "Pizza8", "Pizza9", "Pizza10", "Pizza11", "Pizza12", "Pizza13", "Pizza14", "Pizza15", "Pizza16", "Pizza17", "Pizza18", "Pizza19", "Pizza20", "Pizza21", "Pizza22", "Pizza23", "Pizza24", "Pizza25", "Pizza26", "Pizza27", "Pizza28", "Pizza29", "Pizza30"
+   
+    let pizza = [PizzaModel(name: "Маргарита", image: "1", descriptionPizza: "Сыр Моцарелла, Томатный соус"),
```
pizzas/pizzasList/pizzasArray

KMNervanin commented on RestaurantsApp/ViewController.swift:
```diff
@@ -9,13 +9,26 @@ import UIKit
     
     var tableView = UITableView()
-    
+    let customImageView = UIImageView()
```
зачем это?

sorrymother committed:
LayoutPizzaCells
